ui/components/DetailModal.tsx

```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import styled from "styled-components";
+import { DialogYamlView, YamlViewProps } from "./YamlView";
+
+export enum DetailOptions {
+  YamlView = "YamlView",
+}
+
+export type PropOptions = YamlViewProps;
+
+export type DetailViewProps = {
+  className?: string;
+  component: DetailOptions;
+  props: PropOptions;
+};
+
+function DetailModal({ className, props, component }: DetailViewProps) {
+  switch (component) {
+    case DetailOptions.YamlView:
```

Comment:
I think the `default` case here needs to return `null`. I am worried that when the modal gets closed, there is still something being rendered. If we don't return `null` when the `Drawer` is closed, old elements will appear there momentarily.

This also prevents weird state issues, because every time the `Drawer` gets opened, a new component will be instantiated.

Reply:
So I see below that the `Drawer` will get added/removed from the DOM. I wonder what happens when I go from one Yaml to another without closing it? Can I do that?

Reply:
Totally fine returning null - the rest of the app is blocked when this modal is open - clicking anywhere outside of it will close the modal. Right now there's now way to navigate within the modal.